src/Analysis/Engine/Test/AnalysisTest.cs

```diff
@@ -703,9 +703,12 @@ import mod1
             using (var server = await CreateServerAsync(PythonVersions.LatestAvailable3X)) {
                 var uri1 = TestData.GetTempPathUri("mod1.py");
                 var uri2 = TestData.GetTempPathUri("mod2.py");
-                await server.SendDidOpenTextDocument(uri1, text1);
-                await server.SendDidOpenTextDocument(uri2, text2);
+                using (FileLoading()) {
+                    await server.SendDidOpenTextDocument(uri1, text1);
+                    await server.SendDidOpenTextDocument(uri2, text2);
+                }
 
+                await server.WaitForCompleteAnalysisAsync(CancellationToken.None);
```

Comment:
Ok, these changes make tests pass, but when we make changes to `LanguageServer` so that it doesn't load all the files in the root directory and subdirectories, these tests will become irrelevant to the product behavior, cause there will be no waiting.

Reply:
Even then the analysis will still have to be completed fully on the loaded set and importing mechanism will be loading files that are referenced, even if they are not open.